resources/wikia/modules/pageviewsInSession.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Pageviews in session counter
+ *
+ * What is definition of session in this case?
+ * Session is series of page views (across one or many wikis) done by one user.
+ * It starts when user first enter wiki page and ends when user close last browser tab with wiki page.
+ *
+ * One case is not covered in this solution:
+ * User is on wiki page, navigate to external service and then type wiki address into browser address bar.
+ * When the url is typed in this bar we don't have information about history and last page.
+ */
+define('wikia.pageviewsInSession',
+	['wikia.window', 'wikia.document', 'wikia.cookies', 'mw'], function (window, document, cookies, mw) {
+	'use strict';
+
+	var wikiaDomain = mw.config.get('wgDevelEnvironment') ? '.wikia-dev.com' : '.wikia.com',
+		pageviewsCookieName = 'pageviewsInSession',
+		opentabsNumberCookieName = 'openTabsNumber';
+
+	function init() {
+		increaseNumberOfOpenTabs();
+		setPageviewsCount();
+
+		window.addEventListener('unload', function () {
+			decreaseNumberOfOpenTabs();
+			sessionStorage.setItem('wasWikiPageShownInCurrentTab', true);
+		});
+	}
+
+	function getPageviewsCount() {
+		var pageviews = cookies.get(pageviewsCookieName);
+		return pageviews ? parseInt(pageviews) : 0;
+	}
+
+	function setPageviewsCount() {
+		if (isUserFirstPageview()) {
+			/**
+			 * We need to clear page views when user close all tabs with wiki pages (but not whole browser)
+			 * and then come back to wiki page. Then we want to start another session.
+			 */
+			clearPageviewsCount();
+		}
+
+		cookies.set(pageviewsCookieName, getPageviewsCount() + 1, { domain: wikiaDomain });
+	}
+
+	function clearPageviewsCount() {
+		cookies.set(pageviewsCookieName);
```

Comment:
Does this work without setting the valid domain and path? Since otherwise it will use the default when clearing (which just sets the cookie expiry to `-1`).